wordpress-mobile/WordPress-Android
# Add report user option in Reader post menu

Fixes #17602

This PR adds functionality to report the author of a post from the Reader post `⋮` overflow menu.

- Internal BE ref: D94418-code

> The new option opens the form to report abuse on WP.com and passes the post author info. This form will be updated to support reporting users/authors.

This PR also changes the report icons (see screenshot) for:
- The existing `Report this post` functionality
- The added `Report this user` functionality

## To test:

### 1. Test Report user from Discover
1. Open Reader
2. Tap on the `⋮` overflow menu of one of the posts
   - **Verify** that the report actions have a flag icon
4. Select `Report this user`
5. **Expect** To see a page for reporting the author
   > **Remark** Currently it opens the page to report post/site, this should change.

<details>
<summary><h3>2. Test report user from site post list</h3> (expand)</summary>

1. Open Reader
2. Tap on the header of one post in `Discover` to open the post list of that site
3. Tap on the `⋮` overflow menu of one of the posts
4. Select `Report this user`
6. **Expect** To see a page for reporting the author
   > **Remark** Currently it opens the page to report post/site, this should change.

</details>
<details>
<summary><h3>3. Test report user from post detail screen</h3> (expand)</summary>

1. Open Reader
2. Tap on the excerpt (text above icons) of one post in `Discover` to open the post in reader
3. Tap on the `⋮` overflow menu at the top right of the screen
4. Select `Report this user`
5. **Expect** To see a page for reporting the author
   > **Remark** Currently it opens the page to report post/site, this should change.

</details>

### Screenshot

| Followed Sites | Unfollwed Sites |
| - | - |
| <img width="250" src="https://user-images.githubusercontent.com/4588074/205695226-cbafd68b-1200-42eb-b5a2-68a86fe0797d.png"> | <img width="250" src="https://user-images.githubusercontent.com/4588074/205696504-03ddb72f-8364-4f25-a7aa-6095c17b16ac.png"> |

<details>
<summary><b>Report Author form</b> (web)</summary>

| ![report_author_form](https://user-images.githubusercontent.com/4588074/205920585-df7f2fec-d3cb-416a-94ef-6027d220b889.png) |
| - |
</details>

## Regression Notes
1. Potential unintended areas of impact
   N/a

7. What I did to test those areas of impact (or what existing automated tests I relied on)
   Manual testing.

8. What automated tests I added (or what prevented me from doing so)
   Added test to verify the click action and the presence of the new option.

PR submission checklist:
